### PR TITLE
completed view all grades | Patrick

### PR DESCRIPTION
**Pull request by patrick-delrosario**

### CODE REVIEW

**Comment by patrick-delrosario** [comment]
image

File: components/grade-table.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    for (var i = 0; i < grades.length; i++) {
+      var row = document.createElement("tr");
+      var fullName = document.createElement("td");
+      var course = document.createElement("td");
+      var grade = document.createElement("td");
+      fullName.textContent = grades[i].name;
+      course.textContent = grades[i].course;
+      grade.textContent = grades[i].grade;
+
+      $("tbody").append(row, fullName, course, grade);
```
Comment (scbowler) [inline components/grade-table.js]: Here everything is done in "core" javascript except line #16. You usually don't want to mix core and jQuery you should stick to one or the other. Also you already have a pointer to the `tbody` on line 6 so on line 16 you should't re-query the DOM with jQuery to get another reference to the tbody element. You should change line 16 to use core JavaScript and utilize the `tbody` var you already created above. Look into [appendChild](https://developer.mozilla.org/en-US/docs/Web/API/Node/appendChild) or [append](https://developer.mozilla.org/en-US/docs/Web/API/ParentNode/append)

**Review by scbowler — CHANGES_REQUESTED** [review]
Looking good and you are really close, just need a little cleanup in your `GradeTable` class. See below comment.

File: components/apps.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
+class App {
+  handleGetGradesError(error) {
+    console.error("error");
+  }
+
+  handleGetGradesSuccess(grades) {
+    this.gradeTable.updateGrades(grades);
+  }
+
+  constructor (gradeTable) {
```
Comment (WhiskeyJack96) [inline components/apps.js]: Usually the constuctor should be the first method in a class. You dont need to change it now, but before the next pr you should change it

**Review by WhiskeyJack96 — APPROVED** [review]

**Pull request merged**